From c9c9af3db7e75f376de049f0ec2bd7eca8069d1b Mon Sep 17 00:00:00 2001
From: md_5 <git@md-5.net>
Date: Fri, 28 Jun 2013 19:52:54 +1000
Subject: [PATCH] Disallow Interaction With Self


diff --git a/src/main/java/net/minecraft/server/network/PlayerConnection.java b/src/main/java/net/minecraft/server/network/PlayerConnection.java
index 6a62e51cb..93274196b 100644
--- a/src/main/java/net/minecraft/server/network/PlayerConnection.java
+++ b/src/main/java/net/minecraft/server/network/PlayerConnection.java
@@ -2556,6 +2556,13 @@ public class PlayerConnection extends ServerCommonPacketListenerImpl implements
         if (this.player.hasClientLoaded()) {
             final WorldServer worldserver = this.player.level();
             final Entity entity = packetplayinuseentity.getTarget(worldserver);
+            // Spigot Start
+            if ( entity == player && !player.isSpectator() )
+            {
+                disconnect( IChatBaseComponent.literal( "Cannot interact with self!" ) );
+                return;
+            }
+            // Spigot End
 
             this.player.resetLastActionTime();
             this.player.setShiftKeyDown(packetplayinuseentity.isUsingSecondaryAction());
-- 
2.51.0

