From a0acf0f28f8370eb39734079f5fe9cbbaa86475d Mon Sep 17 00:00:00 2001
From: DerFlash <bte@freenet.de>
Date: Tue, 9 Jul 2013 00:11:12 +0200
Subject: [PATCH] Save ticks lived to nbttag


diff --git a/src/main/java/net/minecraft/world/entity/Entity.java b/src/main/java/net/minecraft/world/entity/Entity.java
index 04a956fe2..686da54a1 100644
--- a/src/main/java/net/minecraft/world/entity/Entity.java
+++ b/src/main/java/net/minecraft/world/entity/Entity.java
@@ -2414,6 +2414,7 @@ public abstract class Entity implements SyncedDataHolder, DebugValueSource, INam
             if (maxAirTicks != getDefaultMaxAirSupply()) {
                 valueoutput.putInt("Bukkit.MaxAirSupply", getMaxAirSupply());
             }
+            valueoutput.putInt("Spigot.ticksLived", this.tickCount);
             // CraftBukkit end
             valueoutput.storeNullable("CustomName", ComponentSerialization.CODEC, this.getCustomName());
             if (this.isCustomNameVisible()) {
@@ -2538,6 +2539,11 @@ public abstract class Entity implements SyncedDataHolder, DebugValueSource, INam
             }
 
             // CraftBukkit start
+            // Spigot start
+            if (this instanceof EntityLiving) {
+                this.tickCount = valueinput.getIntOr("Spigot.ticksLived", this.tickCount);
+            }
+            // Spigot end
             this.persist = valueinput.getBooleanOr("Bukkit.persist", this.persist);
             this.visibleByDefault = valueinput.getBooleanOr("Bukkit.visibleByDefault", this.visibleByDefault);
             // SPIGOT-6907: re-implement LivingEntity#setMaximumAir()
-- 
2.52.0

