From c5a21f7bae7d8e4d86c6d7edaeceacb79aaec309 Mon Sep 17 00:00:00 2001
From: DerFlash <bte@freenet.de>
Date: Tue, 9 Jul 2013 00:11:12 +0200
Subject: [PATCH] Save ticks lived to nbttag


diff --git a/src/main/java/net/minecraft/world/entity/Entity.java b/src/main/java/net/minecraft/world/entity/Entity.java
index 3f15eeb33..0b91ac8f1 100644
--- a/src/main/java/net/minecraft/world/entity/Entity.java
+++ b/src/main/java/net/minecraft/world/entity/Entity.java
@@ -2489,6 +2489,7 @@ public abstract class Entity implements Nameable, EntityAccess, ScoreHolder, Syn
             if (maxAirTicks != getDefaultMaxAirSupply()) {
                 output.putInt("Bukkit.MaxAirSupply", getMaxAirSupply());
             }
+            output.putInt("Spigot.ticksLived", this.tickCount);
             // CraftBukkit end
             output.storeNullable("CustomName", ComponentSerialization.CODEC, this.getCustomName());
             if (this.isCustomNameVisible()) {
@@ -2613,6 +2614,11 @@ public abstract class Entity implements Nameable, EntityAccess, ScoreHolder, Syn
             }
 
             // CraftBukkit start
+            // Spigot start
+            if (this instanceof LivingEntity) {
+                this.tickCount = input.getIntOr("Spigot.ticksLived", this.tickCount);
+            }
+            // Spigot end
             this.persist = input.getBooleanOr("Bukkit.persist", this.persist);
             this.visibleByDefault = input.getBooleanOr("Bukkit.visibleByDefault", this.visibleByDefault);
             // SPIGOT-6907: re-implement LivingEntity#setMaximumAir()
-- 
2.54.0

