From 0d3f394817639312520924deda3f2180cdceda71 Mon Sep 17 00:00:00 2001
From: drXor <mcyoungsota@gmail.com>
Date: Fri, 25 Apr 2014 18:17:30 -0400
Subject: [PATCH] Configurable UserCache cap


diff --git a/src/main/java/net/minecraft/server/players/CachedUserNameToIdResolver.java b/src/main/java/net/minecraft/server/players/CachedUserNameToIdResolver.java
index 94adc65f4..20ffab397 100644
--- a/src/main/java/net/minecraft/server/players/CachedUserNameToIdResolver.java
+++ b/src/main/java/net/minecraft/server/players/CachedUserNameToIdResolver.java
@@ -187,7 +187,7 @@ public class CachedUserNameToIdResolver implements UserNameToIdResolver {
         JsonArray jsonarray = new JsonArray();
         DateFormat dateformat = createDateFormat();
 
-        this.getTopMRUProfiles(1000).forEach((cachedusernametoidresolver_a) -> {
+        this.getTopMRUProfiles(org.spigotmc.SpigotConfig.userCacheCap).forEach((cachedusernametoidresolver_a) -> { // Spigot
             jsonarray.add(writeGameProfile(cachedusernametoidresolver_a, dateformat));
         });
         String s = this.gson.toJson(jsonarray);
diff --git a/src/main/java/org/spigotmc/SpigotConfig.java b/src/main/java/org/spigotmc/SpigotConfig.java
index f20e896f2..d8cd8b47a 100644
--- a/src/main/java/org/spigotmc/SpigotConfig.java
+++ b/src/main/java/org/spigotmc/SpigotConfig.java
@@ -314,4 +314,10 @@ public class SpigotConfig
         replaceCommands = new HashSet<String>( (List<String>) getList( "commands.replace-commands",
                 Arrays.asList( "setblock", "summon", "testforblock", "tellraw" ) ) );
     }
+
+    public static int userCacheCap;
+    private static void userCacheCap()
+    {
+        userCacheCap = getInt( "settings.user-cache-size", 1000 );
+    }
 }
-- 
2.51.0

