From a55eb6400b77637da824f76578dbe6689e80d039 Mon Sep 17 00:00:00 2001
From: ninja <xninja@openmailbox.org>
Date: Tue, 8 Apr 2014 14:05:19 +0200
Subject: [PATCH] Implement PlayerSpawnLocationEvent.


diff --git a/src/main/java/net/minecraft/server/network/config/PrepareSpawnTask.java b/src/main/java/net/minecraft/server/network/config/PrepareSpawnTask.java
index 5e30fe820..f90cf07dd 100644
--- a/src/main/java/net/minecraft/server/network/config/PrepareSpawnTask.java
+++ b/src/main/java/net/minecraft/server/network/config/PrepareSpawnTask.java
@@ -32,6 +32,11 @@ import org.slf4j.Logger;
 import net.minecraft.resources.ResourceKey;
 import net.minecraft.world.level.World;
 // CraftBukkit end
+// Spigot start
+import org.bukkit.Location;
+import org.bukkit.craftbukkit.CraftWorld;
+import org.bukkit.entity.Player;
+// Spigot end
 
 public class PrepareSpawnTask implements ConfigurationTask {
 
@@ -224,6 +229,18 @@ public class PrepareSpawnTask implements ConfigurationTask {
                 Objects.requireNonNull(entityplayer);
                 optional.ifPresent(entityplayer::load);
                 entityplayer.snapTo(this.spawnPosition, this.spawnAngle.x, this.spawnAngle.y);
+                // Spigot start - spawn location event
+                Player spawnPlayer = entityplayer.getBukkitEntity();
+                org.spigotmc.event.player.PlayerSpawnLocationEvent ev = new org.spigotmc.event.player.PlayerSpawnLocationEvent(spawnPlayer, spawnPlayer.getLocation());
+                spawnPlayer.getServer().getPluginManager().callEvent(ev);
+
+                Location loc = ev.getSpawnLocation();
+                WorldServer spawnWorld = ((CraftWorld) loc.getWorld()).getHandle();
+
+                entityplayer.spawnIn(spawnWorld, true);
+                entityplayer.gameMode.setLevel((WorldServer) entityplayer.level());
+                entityplayer.absSnapTo(loc.getX(), loc.getY(), loc.getZ(), loc.getYaw(), loc.getPitch());
+                // Spigot end
                 PrepareSpawnTask.this.server.getPlayerList().placeNewPlayer(networkmanager, entityplayer, commonlistenercookie);
                 optional.ifPresent((valueinput) -> {
                     entityplayer.loadAndSpawnEnderPearls(valueinput);
-- 
2.51.0

